iphone/Maps/UI/PlacePage/MWMPlacePageData.mm

```diff
@@ -367,69 +364,67 @@ - (void)fillOnlineBookingSections
           if (!self || hotelId != hotelInfo.m_hotelId)
             return;
 
-          dispatch_async(dispatch_get_main_queue(), [self, hotelInfo] {
-            m_hotelInfo = hotelInfo;
+          m_hotelInfo = hotelInfo;
 
-            auto & sections = self->m_sections;
-            auto const begin = sections.begin();
-            auto const end = sections.end();
+          auto & sections = self->m_sections;
+          auto const begin = sections.begin();
+          auto const end = sections.end();
 
-            NSUInteger const position = find(begin, end, Sections::Bookmark) != end ? 2 : 1;
-            NSUInteger length = 0;
-            auto it = m_sections.begin() + position;
+          NSUInteger const position = find(begin, end, Sections::Bookmark) != end ? 2 : 1;
+          NSUInteger length = 0;
+          auto it = m_sections.begin() + position;
 
-            if (!hotelInfo.m_photos.empty())
-            {
-              it = sections.insert(it, Sections::HotelPhotos) + 1;
-              m_hotelPhotosRows.emplace_back(HotelPhotosRow::Regular);
-              length++;
-            }
+          if (!hotelInfo.m_photos.empty())
+          {
+            it = sections.insert(it, Sections::HotelPhotos) + 1;
+            m_hotelPhotosRows.emplace_back(HotelPhotosRow::Regular);
+            length++;
+          }
 
-            if (!hotelInfo.m_description.empty())
-            {
-              it = sections.insert(it, Sections::HotelDescription) + 1;
-              m_hotelDescriptionRows.emplace_back(HotelDescriptionRow::Regular);
-              m_hotelDescriptionRows.emplace_back(HotelDescriptionRow::ShowMore);
-              length++;
-            }
+          if (!hotelInfo.m_description.empty())
+          {
+            it = sections.insert(it, Sections::HotelDescription) + 1;
+            m_hotelDescriptionRows.emplace_back(HotelDescriptionRow::Regular);
+            m_hotelDescriptionRows.emplace_back(HotelDescriptionRow::ShowMore);
+            length++;
+          }
 
-            auto const & facilities = hotelInfo.m_facilities;
-            if (!facilities.empty())
+          auto const & facilities = hotelInfo.m_facilities;
+          if (!facilities.empty())
+          {
+            it = sections.insert(it, Sections::HotelFacilities) + 1;
```

Comment:
Am I right that it and length are updated together? If so, how about to create a short function, say:
```c++
template <typename It>
void InsertToSection(Sections & sections, Section section, It & it, size_t & length)
{
  it = sections.insert(it, section) + 1;
  ++length;
}
```